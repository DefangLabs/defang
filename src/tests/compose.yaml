name: tests
services:
  dfnx:
    build:
      context: .
      dockerfile: Dockerfile
      args:
        DNS: "${COMPOSE_PROJECT_NAME}-dfnx.prod1.defang.dev"
    env_file:
      - fileName.env
    deploy:
      resources:
        limits:
          cpus: '0.50'
          memory: 512M
        reservations:
          cpus: '0.25'
          memory: 256M
    ports:
      - target: 80
        mode: ingress
      - target: 1234
        # mode: host
    secrets:
      - dummy
    healthcheck:
      test: ["CMD", "curl", "-f", "http://localhost/"]
      # disable: true

  # dfnx:
  #   build:
  #     context: .
      # dockerfile: Dockerfile.dfn
    # ports:
      # - 80

  echo:
    image: ealen/echo-server
    ports:
      - target: 80
        mode: ingress
    healthcheck:
      test: ["CMD", "curl", "-f", "http://localhost/"]
    # domainname: echotest.gnafed.click
    profiles:
      - donotstart

secrets:
  dummy:
    external: true
    name: dummy
