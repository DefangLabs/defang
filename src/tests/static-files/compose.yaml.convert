[
  {
    "name": "x",
    "image": "blah",
    "platform": 2,
    "internal": true,
    "static_files": {
      "folder": "./folder"
    },
    "networks": 1
  },
  {
    "name": "y",
    "image": "blah",
    "platform": 2,
    "internal": true,
    "static_files": {
      "folder": "./folder"
    },
    "networks": 1
  }
]