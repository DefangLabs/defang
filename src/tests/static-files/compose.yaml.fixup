[
  {
    "command": null,
    "entrypoint": null,
    "image": "blah",
    "networks": {
      "default": null
    }
  },
  {
    "command": null,
    "entrypoint": null,
    "image": "blah",
    "networks": {
      "default": null
    }
  }
]