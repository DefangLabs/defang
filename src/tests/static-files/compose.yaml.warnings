 ! service "x": missing compose directive: restart; assuming 'unless-stopped' (add 'restart' to silence)
 ! service "x": missing memory reservation; specify deploy.resources.reservations.memory to avoid out-of-memory errors
 ! service "y": missing compose directive: restart; assuming 'unless-stopped' (add 'restart' to silence)
 ! service "y": missing memory reservation; specify deploy.resources.reservations.memory to avoid out-of-memory errors
 ! service "y": unsupported compose extension: "x-unsupported"