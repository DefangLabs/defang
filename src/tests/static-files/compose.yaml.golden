name: static-files
services:
  x:
    image: blah
    networks:
      default: null
    x-defang-static-files: ./folder
  "y":
    image: blah
    networks:
      default: null
    x-defang-static-files:
      folder: ./folder
      redirects: true - www.example.com
    x-unsupported: asdf
networks:
  default:
    name: static-files_default
